### ISSUE TITLE
Add support for Linux-specific UA string

### ISSUE THREAD

**Issue by larsbergstrom** [servo/servo#7158]
In #7143, we added a User Agent string for Servo, but only for Android, Gonk, and OSX. We would like to add a platform-specific Linux modifier. 

Based on information from @gerv in https://github.com/servo/servo/pull/7143#issuecomment-129859641, we will need to include the full set of bits such as `X11; Ubuntu; Linux x86_64`, which will require some analysis of the environment at startup. These additional bits (distro name) appear to be used for sites that want to be able to suggest the correct download.

CC: @SimonSapin

**Comment by SimonSapin** [comment]
Is the distro name really needed? My Firefoxes, respectively Release from archlinux.org/packages and Nightly from mozilla.org, currently have:

```
Mozilla/5.0 (X11; Linux x86_64; rv:39.0) Gecko/20100101 Firefox/39.0
Mozilla/5.0 (X11; Linux x86_64; rv:42.0) Gecko/20100101 Firefox/42.0
```

I wouldn’t be surprised if `Ubuntu` is a vanity bit added by packagers.

**Comment by larsbergstrom** [comment]
@SimonSapin That seems consistent with https://wiki.mozilla.org/Firefox/User_Agent, though there isn't a ton of info I can find there or in other bugs on BMO.

According to the source (http://mxr.mozilla.org/mozilla-central/source/netwerk/protocol/http/nsHttpHandler.cpp#695), the `X11` is a historical bit and not used by sites for anything.

**Comment by SimonSapin** [comment]
https://bazaar.launchpad.net/~mozillateam/firefox/firefox.wily/view/head:/debian/patches/ubuntu-ua-string-changes.patch is an Ubuntu-specific patch adding the `Ubuntu` bit. The commit message that first added it says "Modify the UA string to add "Ubuntu" to the platform component", but doesn’t say why.

**Comment by gerv** [comment]
Yes, don't include the "Ubuntu" bit. Copy what stock Firefox does, which is "X11; Linux x86_64" or some related string for 32-bit if you guys support that.

**Comment by larsbergstrom** [comment]
Marking this as `E-Easy`, as we can skip the runtime determinination and figure out all of this statically from the compilation target architecture.

**Comment by jdm** [comment]
See `default_user_agent_string` in `opts.rs`.

**Comment by jdm** [comment]
Fixed by #8353.
